curvefs/src/metaserver/s3/metaserver_s3.h

```diff
@@ -36,6 +36,9 @@ class S3Client {
     virtual void Init(const curve::common::S3AdapterOption& option) = 0;
     virtual int Delete(const std::string& name) = 0;
     virtual int DeleteBatch(const std::list<std::string>& nameList) = 0;
+    virtual void Reinit(const std::string& ak, const std::string& sk,
```

Comment:
it's a superset of s3client in mds_s3.h, you should reduce duplication.

Reply:
> it's a superset of s3client in mds_s3.h, you should reduce duplication.

this is different from mds_s3, mds_s3 work for check s3 info available, this work for delete s3 object. 
There is no inheritance relationship between these two s3.